File: div7.cpp
// Source: https://usaco.guide/general/io

#include <bits/stdc++.h>
#include <map>
using namespace std;

int main() {
	ifstream fin ("div7.in");
  	ofstream fout ("div7.out");
	int n; fin >> n;
	int psum[n+1];
	psum[0] = 0;
	for (int i = 0; i < n; i++) {
		int x; fin >> x;
		psum[i+1] = (x + psum[i]) % 7;
	}

	int ans = 0;
	map<int, int> found;
	for (int i = 1; i < n+1; i++) {
		int j = psum[i];
		if (found.count(j)==1) {
			int diff = i - found[j];
			ans = max(ans, diff);
		} else {
			found[j] = i;
		}
	}
	fout << ans << "\n";
}

File: div7.py
import sys
sys.setrecursionlimit(1000000)

sys.stdin = open("div7.in","r")
sys.stdout = open("div7.out","w")

n = int(input())
psum = [0]
for i in range(n):
    j = int(input())
    psum.append((j+psum[-1])%7)
# print(psum[1:])


s = 0
found = {}
for i,j in enumerate(psum[1:]):
    if j in found:
        s = max(s, i - found[j])
    else:
        found[j] = i
print(s)
